handlers/whatsapp/whatsapp.go

```diff
@@ -600,7 +605,88 @@ func (h *handler) SendMsg(ctx context.Context, msg courier.Msg) (courier.MsgStat
 	return status, nil
 }
 
-func sendWhatsAppMsg(msg courier.Msg, sendPath *url.URL, token string, payload interface{}) (string, string, []*courier.ChannelLog, error) {
+// fetchMediaID tries to fetch the id for the uploaded media, setting the result in redis.
+func (h *handler) fetchMediaID(msg courier.Msg, mimeType, mediaURL string) (string, []*courier.ChannelLog, error) {
+	var logs []*courier.ChannelLog
+	start := time.Now()
+
+	// check on cache first
+	rc := h.Backend().RedisPool().Get()
+	defer rc.Close()
+
+	cacheKey := fmt.Sprintf("whatsapp_media:%s:%s", msg.Channel().UUID().String(), mediaURL)
```

Comment:
Instead of having one key/value at the top level for each file why don't we group these by channel. It will be more efficient from a memory perspective for Redis and easier to grok if looking at plain keys.

So basically ah HSET with `whatsapp_media:[channel_uuid]`as the set name, `mediaURL` as the key name and the value being the id. Expiring daily.

Reply:
`whatsapp_media` should be a const same with failure key.

Reply:
Do you mean to expire the `whatsapp_media:[channel_uuid]`? As far as I know, Redis only supports TTL at top-level keys.

Reply:
Ya, generally the way we do this is have the date in the key as well so that those keys "roll" every day and then expire out. The main argument for that is that it is both easier to clear out caches for a channel and that it is less memory in Redis because the rather verbose portion of the key `whatsapp_media:uuid-uuid-uuid-uuid-uuid-uuid` is not repeated for every entry.

See: 
  https://instagram-engineering.com/storing-hundreds-of-millions-of-simple-key-value-pairs-in-redis-1091ae80f74c
And:
  https://redis.io/topics/memory-optimization (Use hashes when possible)

Reply:
Great, thanks for the references!